model/validation.go

```diff
@@ -44,9 +45,9 @@ func IsDNS1123Label(value string) bool {
 }
 
 // Validate confirms that the names in the configuration key are appropriate
-func (k *ConfigKey) Validate() error {
+func (k *Key) Validate() error {
 	var errs error
-	if !kindRegexp.MatchString(k.Kind) {
+	if !IsDNS1123Label(k.Kind) {
```

Comment:
Doesn't this mean we need atleast two parts in the kind name ? route-rule.istio.io? If so the constants above need to be updated.